src/scripts/main.js

```diff
@@ -1 +1,73 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  const handler = () => {
+    resolve();
+    document.addEventListener('click', resolve);
+  };
+
+  document.addEventListener('click', handler);
+
+  setTimeout(reject, 3000);
+});
+
+const secondPromise = new Promise((resolve, reject) => {
+  const handler = () => {
+    resolve();
+    document.addEventListener('click', resolve);
+    document.addEventListener('contextmenu', resolve);
+  };
+
+  document.addEventListener('contextmenu', handler);
+  document.addEventListener('click', handler);
+});
+
+const thirdPromise = new Promise((resolve, reject) => {
+  let left = false;
+  let right = false;
+
+  const leftClick = () => {
+    left = true;
+
+    if (right) {
+      resolve();
+      document.addEventListener('click', leftClick);
+      document.addEventListener('contextmenu', rightClick);
+    }
+  };
+
+  const rightClick = () => {
+    right = true;
+
+    if (left) {
+      resolve();
+      document.addEventListener('click', leftClick);
```

Comment:
Same issue here as in line 6. You are adding an event listener inside the handler for the click event. This could potentially lead to a memory leak. Consider removing the event listener once it has been triggered.